### PR TITLE
LoggingSourceGenerator - Fixes InvalidCastException when using other attributes

### LINKED ISSUE #67167: Logging Source Generator fails with a InvalidCastException
### Description

Adding a SuppressMessageAttribute to a method marked with LoggerMessageAttribute results in a failed compilation

### Reproduction Steps

1. Create a new Class Library
2. Add `<PackageReference Include="Microsoft.Extensions.Logging" Version="6.0.1" />`
3. Write the following code in Class1.cs
```c#
using System.Diagnostics.CodeAnalysis;
using Microsoft.Extensions.Logging;

namespace ClassLibrary1
{
    public partial class Class1
    {
        private ILogger _logger;
        public Class1(ILogger logger) => _logger = logger;

        [LoggerMessage(EventId = 1, Level = LogLevel.Error, Message = "Log Message {Name}")]
        [SuppressMessage("ReSharper", "InconsistentNaming")]
        private partial void LogError(string Name, Exception exception);
    }
}
```
4. Build

### Expected behavior

The source generator should generate.

### Actual behavior

The source generator fails with an error in the build log:

CSC : warning CS8785: Generator 'LoggerMessageGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'InvalidCastException' with message 'Unable to cast object of type 'System.String' to type 'System.Int32'.'

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

dotnet SDK version: 6.0.103
OS: Manjaro Linux (x64)

### Other information

_No response_

### PR DESCRIPTION
**Pull request by maryamariyan**
- alongside LoggerMessageAttribute

Fixes #67167

### CODE REVIEW

**Review by tarekgh — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs [hunk 1]
```diff
@@ -141,15 +141,20 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                     }
 
                                     bool hasMisconfiguredInput = false;
-                                    ImmutableArray<AttributeData>? boundAttrbutes = logMethodSymbol?.GetAttributes();
+                                    ImmutableArray<AttributeData>? boundAttributes = logMethodSymbol?.GetAttributes();
 
-                                    if (boundAttrbutes == null)
+                                    if (boundAttributes == null || boundAttributes!.Value.Length == 0)
```
Comment (eerhardt) [inline src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs:146]: Why did you need a `!` here? You just checked it for `null` on the same line.
